llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -5681,6 +6079,150 @@ static SDValue combineADDRSPACECAST(SDNode *N,
   return SDValue();
 }
 
+static SDValue PerformFP_ROUNDCombine(SDNode *N,
+                                      TargetLowering::DAGCombinerInfo &DCI) {
+  SDLoc DL(N);
+  SDValue Op = N->getOperand(0);
+  SDValue Trunc = N->getOperand(1);
+  EVT NarrowVT = N->getValueType(0);
+  EVT WideVT = Op.getValueType();
+
+  // v2[b]f16 = fp_round (v2f32 A)
+  // -> v2[b]f16 = (build_vector ([b]f16 = fp_round (extractelt A, 0)),
+  //                             ([b]f16 = fp_round (extractelt A, 1)))
```

Comment:
Why is this required?

Reply:
Added a comment explaining. `v2f16 = fp_round v2f32` is a legal operation but with two f32 input registers, not a single `v2f32`. So we have to expand this node. A later pattern match will turn it into `cvt.f16x2.f32`.